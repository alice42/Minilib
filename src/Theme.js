export const theme = {
  backgroundColor: '#ECEFF1',
  onBackground: '#000000',
  surfaceColor: '#FFFFFF',
  surfaceColorHover: '#F5F5F5',
  surfaceColorActive: '#E0E0E0',
  onSurface: '#000000',
  primaryColor: '#0277BD',
  primaryColorHover: '#0288D1',
  primaryColorActive: '#01579B',
  onPrimary: '#FFFFFF',
  secondaryColor: '#00897B',
  secondaryColorHover: '#00695C',
  secondaryColorActive: '#004D40',
  onSecondary: '#FFFFFF',
  tertiaryColor: '#EF6C00',
  tertiaryColorHover: '#FB8C00',
  tertiaryColorActive: '#E65100',
  onTertiary: '#FFFFFF',
  fontPrimary: {
    regular: 'Lato,sans-serif',
    bold: 'Lato,sans-serif',
    light: 'Lato,sans-serif'
  },
  fontSecondary: {
    regular: 'arial,sans-serif',
    bold: 'arial,sans-serif',
    light: 'arial,sans-serif'
  },
  fontTertiary: {
    regular: 'Georgia,serif',
    bold: 'Georgia,serif',
    light: 'Georgia,serif'
  },
  font48: '4rem',
  font43: '3.6rem',
  font38: '3.2rem',
  font34: '2.9rem',
  font30: '2.5rem',
  font29: '2.4rem',
  font28: '2.3rem',
  font26: '2.1rem',
  font24: '2rem',
  font23: '1.9rem',
  font22: '1.85rem',
  font20: '1.65rem',
  font19: '1.6rem',
  font18: '1.5rem',
  font17: '1.4rem',
  font16: '1.35rem',
  font15: '1.25rem',
  font14: '1.2rem',
  font13: '1.1rem',
  font12: '1rem',
  font11: '0.95rem',
  font10: '0.9rem',
  font9: '0.75rem',
  font8: '0.7rem',
  font6: '0.5rem',
  overlayOpacity: 0.7,
  borderRadius: '5px',
  border: 'solid 5px #000000'
}
