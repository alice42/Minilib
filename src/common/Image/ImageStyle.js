import styled, { css } from 'styled-components'

const imageDefined = () => css`
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ height }) => height && css`
    height: ${height};
  `}
  ${({ width }) => width && css`
    width: ${width};
  `}
  ${({ rounded, theme }) => rounded && css`
    border-radius: ${typeof rounded === 'string' ? rounded : theme.borderRadius};;
  `}
  ${({ border, theme }) => border && css`
    box-sizing: border-box;
    border: ${typeof border === 'string' ? border : theme.border};
  `}
  filter: ${({ blur }) => 'blur(' + blur + ')'};
`

const ImageStyle = styled.div`
  position: relative;

  ${({ asBackground, mosaic }) => (asBackground || mosaic) && css`
    & > div:first-child {
      ${imageDefined}
      background-position: ${mosaic ? 'left top' : 'center'};
      /* Set Image as background when asBackground=true */
      ${({ src }) => css`background-image: url(${src});`}
      ${({ cover, contain, filled }) => css`
        background-size: ${mosaic ? (typeof mosaic === 'string' ? mosaic : '100px 100px') : filled ? '100% 100%' : contain ? 'contain' : cover ? 'cover' : 'auto'};
      `}
      /* Set background-repeat when mosaic=true */
      background-repeat: ${mosaic ? 'repeat' : 'no-repeat'};
    }
  `}

  & > img {
    ${imageDefined}
    object-fit: ${({ cover, contain, filled }) => filled ? 'fill' : contain ? 'contain' : cover ? 'cover' : 'unset'};
  }
  ${({ customStyle }) => css(customStyle)}
`

export default ImageStyle
