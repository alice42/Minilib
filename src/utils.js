export let conflictsValidator = (validType, conflicts) => (props, propName, componentName) => {
  let conflictedProp
  if (typeof props[propName] !== validType) return new Error(`<${componentName}>: ${propName} must be a ${validType}!`)
  if (props[propName] && (conflictedProp = conflicts.find(p => props[p]))) {
    return new Error(`<${componentName}>: ${propName} can't be used if ${conflictedProp} is already used!`)
  }
}
