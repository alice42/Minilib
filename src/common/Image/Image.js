import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Overlay from '../Overlay/Overlay'
import ImageStyle from './ImageStyle'
import { conflictsValidator } from '../../utils'

class Image extends Component {
  static propTypes = {
    /** link to the image file ("./picture.jpg") */
    src: PropTypes.string.isRequired,
    /** round borders style on image (ex: true, "5px") */
    rounded: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** object-fit: cover (if :=false => 'none') */
    cover: conflictsValidator('boolean', ['contain', 'filled']),
    /** object-fit: contain */
    contain: conflictsValidator('boolean', ['cover', 'filled']),
    /** object-fit: fill */
    filled: conflictsValidator('boolean', ['contain', 'cover']),
    /** image borders ("solid 5px #000000") */
    border: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** add a blur effect on the image ("10px", "1.2em", ...) */
    blur: PropTypes.string,
    /** height of the image ("200px", "30%", "45vh", ...) */
    height: PropTypes.string,
    /** width of the image ("200px", "30%", "45vw", ...) */
    width: PropTypes.string,
    /** set the image to a background-image */
    asBackground: PropTypes.bool,
    /** asBackground and background-repeat to the image (true, "20px 15vh", ...) */
    mosaic: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    /** pass props to the overlay, <Image>{ overlay content }</Image> */
    overlay: PropTypes.object,
    /** custom style object */
    customStyle: PropTypes.object
  }
  static defaultProps = {
    rounded: false,
    cover: false,
    contain: false,
    filled: false,
    asBackground: false,
    border: false,
    mosaic: false,
    blur: '0px',
    overlay: {},
    customStyle: {}
  }

  render () {
    const { src, title, asBackground, mosaic, overlay } = this.props
    const alt = src.indexOf('/') !== -1 ? src.slice(src.lastIndexOf('/') + 1) : src
    return (
      <ImageStyle {...this.props}>
        {
          asBackground || mosaic ? <div />
            : <img
              src={src}
              alt={alt}
              title={title || (alt.indexOf('.') !== -1 ? alt.slice(0, alt.lastIndexOf('.')) : alt)}
            />
        }
        {
          this.props.children || (overlay && Object.keys(overlay).length) ? <Overlay {...overlay}>{ this.props.children }</Overlay> : null
        }
      </ImageStyle>
    )
  }
}

export default Image
