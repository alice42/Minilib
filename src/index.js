import Button from './common/Buttons/Button'
import Icon from './common/Icons/Icon'
import Image from './common/Image/Image'
import InputText from './common/Inputs/InputText/InputText'
import ExternalLink from './common/Links/ExternalLink'
import Overlay from './common/Overlay/Overlay'
import Video from './common/Video/Video'

export { Button }
export { Icon }
export { Image }
export { InputText }
export { ExternalLink }
export { Overlay }
export { Video }
