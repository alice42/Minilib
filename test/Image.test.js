import React from 'react'
import Image from '../src/common/Image/Image'
import 'jest-styled-components'
import { renderCreateWithThemeJSON } from './TestUtils'
import { theme } from '../src/Theme'

const src = '/image/koala.jpg'

test(':src', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} />)
  ]
  expect(Image).toThrow(Error)
  expect(comps[0].children).not.toBeNull()
  expect(comps[0].children[0].type).toEqual('img')
  expect(comps[0].children[0].props.src).toEqual(src)
  expect(comps[0]).toMatchSnapshot()
})

test(':height & :width', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} height={'200px'} width={'500px'} />),
    renderCreateWithThemeJSON(<Image src={src} height={'200px'} />),
    renderCreateWithThemeJSON(<Image src={src} width={'500px'} />)
  ]
  expect(comps[0]).toHaveStyleRule('height', '200px', { modifier: '& > img' })
  expect(comps[0]).toHaveStyleRule('width', '500px', { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('height', '200px', { modifier: '& > img' })
  expect(comps[1]).not.toHaveStyleRule('width', { modifier: '& > img' })
  expect(comps[2]).toHaveStyleRule('width', '500px', { modifier: '& > img' })
  expect(comps[2]).not.toHaveStyleRule('height', { modifier: '& > img' })
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
})

test(':object-fit', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} />),
    renderCreateWithThemeJSON(<Image src={src} cover />),
    renderCreateWithThemeJSON(<Image src={src} contain />),
    renderCreateWithThemeJSON(<Image src={src} filled />),
    renderCreateWithThemeJSON(<Image src={src} cover={false} />),
    renderCreateWithThemeJSON(<Image src={src} asBackground cover />),
    renderCreateWithThemeJSON(<Image src={src} asBackground contain />),
    renderCreateWithThemeJSON(<Image src={src} asBackground filled />),
    renderCreateWithThemeJSON(<Image src={src} asBackground />)
  ]
  expect(comps[0]).toHaveStyleRule('object-fit', 'unset', { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('object-fit', 'cover', { modifier: '& > img' })
  expect(comps[2]).toHaveStyleRule('object-fit', 'contain', { modifier: '& > img' })
  expect(comps[3]).toHaveStyleRule('object-fit', 'fill', { modifier: '& > img' })
  expect(comps[4]).toHaveStyleRule('object-fit', 'unset', { modifier: '& > img' })
  expect(comps[5]).toHaveStyleRule('background-size', 'cover', { modifier: '& > div:first-child' })
  expect(comps[6]).toHaveStyleRule('background-size', 'contain', { modifier: '& > div:first-child' })
  expect(comps[7]).toHaveStyleRule('background-size', '100% 100%', { modifier: '& > div:first-child' })
  expect(comps[8]).toHaveStyleRule('background-size', 'auto', { modifier: '& > div:first-child' })
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
  expect(comps[3]).toMatchSnapshot()
  expect(comps[4]).toMatchSnapshot()
})

test(':rounded', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} />),
    renderCreateWithThemeJSON(<Image src={src} rounded />),
    renderCreateWithThemeJSON(<Image src={src} rounded={'87px'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('border-radius', { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('border-radius', theme.borderRadius, { modifier: '& > img' })
  expect(comps[2]).toHaveStyleRule('border-radius', '87px', { modifier: '& > img' })
  expect(comps[1]).toMatchSnapshot()
})

test(':asBackground', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} />),
    renderCreateWithThemeJSON(<Image src={src} asBackground />),
    renderCreateWithThemeJSON(<Image src={src} mosaic />),
    renderCreateWithThemeJSON(<Image src={src} mosaic={'50px 50px'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('background-image', { modifier: '& > div:first-child' })
  expect(comps[0]).not.toHaveStyleRule('background-size', { modifier: '& > div:first-child' })
  expect(comps[1]).toHaveStyleRule('background-image', `url(${src})`, { modifier: '& > div:first-child' })
  expect(comps[1]).toHaveStyleRule('background-size', 'auto', { modifier: '& > div:first-child' })
  expect(comps[2]).toHaveStyleRule('background-image', `url(${src})`, { modifier: '& > div:first-child' })
  expect(comps[2]).toHaveStyleRule('background-size', '100px 100px', { modifier: '& > div:first-child' })
  expect(comps[2]).toHaveStyleRule('background-repeat', 'repeat', { modifier: '& > div:first-child' })
  expect(comps[3]).toHaveStyleRule('background-size', '50px 50px', { modifier: '& > div:first-child' })
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
  expect(comps[3]).toMatchSnapshot()
})

test(':filter', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} blur={'5px'} />),
    renderCreateWithThemeJSON(<Image src={src} asBackground blur={'5px'} />)
  ]
  expect(comps[0]).toHaveStyleRule('filter', `blur(5px)`, { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('filter', `blur(5px)`, { modifier: '& > div:first-child' })
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
})

test(':title & :alt', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src} />),
    renderCreateWithThemeJSON(<Image src={src} title={'HelloWorld!'} />),
    renderCreateWithThemeJSON(<Image src={'noSlashAndNoPoint'} />)
  ]
  expect(comps[0].children[0].props.title).toEqual('koala')
  expect(comps[0].children[0].props.alt).toEqual('koala.jpg')
  expect(comps[1].children[0].props.title).toEqual('HelloWorld!')
  expect(comps[1].children[0].props.alt).toEqual('koala.jpg')
  expect(comps[2].children[0].props.alt).toEqual('noSlashAndNoPoint')
  expect(comps[2].children[0].props.title).toEqual('noSlashAndNoPoint')
})

test(':border', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src}>Hello</Image>),
    renderCreateWithThemeJSON(<Image src={src} border>Hello</Image>),
    renderCreateWithThemeJSON(<Image src={src} border={'dashed 2px red'} />)
  ]
  expect(comps[0]).not.toHaveStyleRule('border', { modifier: '& > img' })
  expect(comps[0]).not.toHaveStyleRule('box-sizing', { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('border', theme.border, { modifier: '& > img' })
  expect(comps[1]).toHaveStyleRule('box-sizing', 'border-box', { modifier: '& > img' })
  expect(comps[2]).toHaveStyleRule('border', 'dashed 2px red', { modifier: '& > img' })
  expect(comps[2]).toHaveStyleRule('box-sizing', 'border-box', { modifier: '& > img' })
  expect(comps[1]).toMatchSnapshot()
  expect(comps[2]).toMatchSnapshot()
})

test(':overlay', () => {
  const comps = [
    renderCreateWithThemeJSON(<Image src={src}>Hello</Image>),
    renderCreateWithThemeJSON(<Image src={src} overlay={{ dark: true }} />)
  ]
  expect(comps[0].children.length).toEqual(2)
  expect(comps[1].children.length).toEqual(2)
  expect(comps[0]).toMatchSnapshot()
  expect(comps[1]).toMatchSnapshot()
})
