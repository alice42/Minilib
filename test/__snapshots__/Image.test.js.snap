// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:asBackground 1`] = `
.c0 {
  position: relative;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  background-position: center;
  background-image: url(/image/koala.jpg);
  background-size: auto;
  background-repeat: no-repeat;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <div />
</div>
`;

exports[`:asBackground 2`] = `
.c0 {
  position: relative;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  background-position: left top;
  background-image: url(/image/koala.jpg);
  background-size: 100px 100px;
  background-repeat: repeat;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <div />
</div>
`;

exports[`:asBackground 3`] = `
.c0 {
  position: relative;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  background-position: left top;
  background-image: url(/image/koala.jpg);
  background-size: 50px 50px;
  background-repeat: repeat;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <div />
</div>
`;

exports[`:border 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: solid 5px #000000;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`:border 2`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: dashed 2px red;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:filter 1`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(5px);
  filter: blur(5px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:filter 2`] = `
.c0 {
  position: relative;
}

.c0 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(5px);
  filter: blur(5px);
  background-position: center;
  background-image: url(/image/koala.jpg);
  background-size: auto;
  background-repeat: no-repeat;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(5px);
  filter: blur(5px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <div />
</div>
`;

exports[`:height & :width 1`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 200px;
  width: 500px;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  height="200px"
  src="/image/koala.jpg"
  width="500px"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:height & :width 2`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 200px;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  height="200px"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:height & :width 3`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 500px;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
  width="500px"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:object-fit 1`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:object-fit 2`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: cover;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:object-fit 3`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: contain;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:object-fit 4`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: fill;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:object-fit 5`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:overlay 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #000000;
  background-color: transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`:overlay 2`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 10px;
  color: #ECEFF1;
  background-color: rgba(0,0,0,0.7);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
}

.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
  <div
    className="c1"
  />
</div>
`;

exports[`:rounded 1`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 5px;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;

exports[`:src 1`] = `
.c0 {
  position: relative;
}

.c0 > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: blur(0px);
  filter: blur(0px);
  object-fit: unset;
}

<div
  className="c0"
  src="/image/koala.jpg"
>
  <img
    alt="koala.jpg"
    src="/image/koala.jpg"
    title="koala"
  />
</div>
`;
